build-system/pr-check.js

```diff
@@ -603,8 +608,9 @@ function main() {
       colors.cyan(process.env.BUILD_SHARD),
       '\n');
 
-  if (process.env.TRAVIS_EVENT_TYPE === 'push') {
-    console.log(fileLogPrefix, 'Running all commands on push build.');
+  if (!isTravisPullRequestBuild()) {
```

Comment:
As discovered last week, we should call `runAllCommands` for all non-PR builds. Otherwise, triggered builds and cron builds will attempt to run PR-related checks when a PR doesn't exist.

In addition, the problems you mentioned were:

> Running a triggered build on an old master commit will now re-generate the baseline of visual diffs on Percy to that older commit

This is not true. There is a check in `bundle-size.js` that writes to the repo only when the build type is explicitly a push build. So the check will do nothing during triggered and cron builds.

> Some commands are now executed with the explicit assumption that this is a push build, but a downstream file performs an `if (process.env.TRAVIS_EVENT_TYPE !== 'push')` or `=== 'pull_request'` check and fails due to the new, unexpected value

With the changes in this PR, the only place where `isTravisPushBuild` is used is `bundle-size.js`, where we do not write to the build artifacts repo for non-push builds.

Reply:
Percy changed their builds so that now every build based on `master` creates a new baseline

Reply:
Gotcha. We could go two ways with this:

1) Allow triggered and cron builds to generate a new baseline. It's rare to have to trigger a build, but it's sometimes necessary after an outage of some sort. In this case, we do want to re-establish a baseline.
2) Run visual diff tests in skip mode if the build type isn't one of push or PR.

Reply:
I'd rather do 3. Skip visual diff tests at all if the build type isn't push/PR
If we really need to recreate the baseline, we can do it manually outside of Travis. It's rare enough that I'd rather guard against accidents than have it be built in to the code

Reply:
Done in the latest commit.